### DISCUSSION TITLE
Unable to enable file-based logging inside the Docker container for Keycloak Quarkus 19.0.3

### DISCUSSION THREAD

**Discussion by pallavi0412** [keycloak/keycloak#36131] [Q&A]
### Description

I am trying to set up file-based logging inside a Docker container for Keycloak Quarkus v19.0.3.

I have tried few approaches, including adding the following application.properties configuration for logging:- 

----------------------------------------------------------------------------------------------

quarkus.log.file.enable=true

quarkus.log.file.path=/opt/keycloak/logs/keycloak.log
quarkus.log.file.rotation.size=10M            # Rotate logs when the size exceeds 10 MB
quarkus.log.file.rotation.max-files=5         # Keep up to 5 rotated log files

quarkus.log.category."org.keycloak".level=INFO
quarkus.log.category."org.keycloak.services".level=DEBUG
quarkus.log.category."org.keycloak.events".level=DEBUG

quarkus.log.console.enable=true
quarkus.log.console.level=INFO

quarkus.log.console.json=true

------------------------------------------------------------------------------------------------

I also updated the Dockerfile to copy the application.properties file with the correct permissions - 

------------------------------------------------------------------------------------------------

FROM quay.io/keycloak/keycloak:19.0.3
RUN mkdir -p /opt/keycloak/logs && chmod -R 777 /opt/keycloak/logs 

WORKDIR /opt/keycloak
COPY ./resources/application.properties /opt/keycloak/conf/application.properties

--------------------------------------------------------------------------------------------------

However, I am still unable to generate logs in the file for Keycloak. Has anyone faced a similar issue or could someone help me with the proper steps to enable logging in this setup?

**Comment by shawkins** [comment]
You should try with a later keycloak version. Console logging has first class configuration available, so you don't need to use quarkus.properties.
